### PR TITLE
Handle callable initial_channels

### LINKED ISSUE #66: Callable initital_channels
Initial channels should allow for any callable to be provided, as long as it returns a string or an list/tuple.

### PR DESCRIPTION
**Pull request by bsquidwrd**
* **Please check if the PR fulfills these requirements**
- [x] Tests have been conducted on the PR
- [ ] Docs have been added / updated

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Closes #66 

* **What is the current behavior?** (You can also link to an open issue here)
Callable is not usable for `initial_channels`

* **What is the new behavior (if this is a feature change)?**
Checks if `initial_channels` is a callable, if so invoke it and ensure the new `_initial_channels` variable is set to a list if the function doesn't return a function or tuple

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Not that I'm aware of

* **Other information**:
I hope I did this right 😓

### CODE REVIEW

File: twitchio/websocket.py [hunk 1]
```diff
@@ -75,6 +75,15 @@ def __init__(self, **kwargs):
         self.nick = kwargs.get('nick').lower()
         self.modes = kwargs.pop('modes', ("commands", "tags", "membership"))
         self._initial_channels = kwargs.get('initial_channels')
+        
+        if callable(self._initial_channels):
+            _temp_initial_channels = self._initial_channels()
+            # If the result of the function is a list or tuple, assign it
+            # Otherwise turn it into a list and assign it
```
Comment (scragly) [inline twitchio/websocket.py]: These aren't necessary as the code is self explanatory regarding it's actions.
```suggestion
```

File: twitchio/websocket.py [hunk 2]
```diff
@@ -75,6 +75,15 @@ def __init__(self, **kwargs):
...
+            if type(_temp_initial_channels) == list or type(_temp_initial_channels) == tuple:
```
Comment (scragly) [inline twitchio/websocket.py]: Type checking should be done with isinstance as it's cleaner and doesn't fail on inherited types.

```suggestion
            if isinstance(_temp_initial_channels, (list, tuple)):
```

File: twitchio/websocket.py (hunk 1 above)
Comment (bsquidwrd) [reply twitchio/websocket.py]: Resolved with latest commit on branch

File: twitchio/websocket.py (hunk 2 above)
Comment (bsquidwrd) [reply twitchio/websocket.py]: Resolved with latest commit on branch, didn't realize this would let me auto commit requested changes 😞

**Pull request merged**